src/persist-types/src/codec_impls.rs

```diff
@@ -45,8 +49,8 @@ impl Codec for String {
         buf.extend(self.as_bytes())
     }
 
-    fn decode<'a>(buf: &'a [u8]) -> Result<Self, String> {
-        String::from_utf8(buf.to_owned()).map_err(|err| err.to_string())
+    fn decode<'a>(buf: &'a [u8]) -> Result<Self, CodecError> {
+        String::from_utf8(buf.to_owned()).map_err(|err| CodecError::from(err.to_string()))
```

Comment:
Same here for avoiding `map_err` with a `impl From< std::str::Utf8Error> for CodecError` implementation